Complete sixth sprint

uncledeni opened the pull request:
Make e2e test for creating order in Cypress; 
Make fixture json files for moc test requests;

Make unit tests for all reducers;

Prepare project for deploy in GitHub Pages;
Add link to deployed project in README.md;

add some @ts-ignore for successfull deploy; will fix them later;

uncledeni committed:
Start e2e and unit  tests coverage

make new unit tests for fucntions in 'calculating.ts';
implement 'cypress' for e2e testing;

update cypress config for project;
make new e2e first test;
make moc data files for 'order' and 'user';

uncledeni committed:
Make some updates in testing

make 'prepare' in 'commands.ts' for Cypress;
add 'data-testid' to elements on 'sign-in' and 'modal';
make 'login' response mock;
make 'ingredients' response mock;
update 'spec.cy.ts';

uncledeni committed:
Update cypress test and mock files

uncledeni committed:
Update cypress test and mock files

uncledeni committed:
Cypress testing incomplete; start redux testing

uncledeni committed:
Fix Cypress (incomplete)

Fix Cypress test (still have trouble with token refresh);
update 'fetchWithRefresh' error message check;

uncledeni committed:
make unit test for 'order-details.ts' reducer

make 'index.ts' file for all actions;
export initial state from 'order-details.ts' reducer;

uncledeni committed:
Continue unit tests for redux

update actions 'index.ts' : specify 'burgerIngredientsActions', 'feedOrderDetailsActions' 'ingredientDetailsActions', 'orderDetailsActions';
update 'order-details' redux tesing;
make 'ingredient-details' redux testing;
make 'burger-ingredients' redux testing;
make 'feed-order-details' redux tesing (incomplete);
export initial states from 'burger-ingredients', 'feed-order-details', 'ingredient-details';
update 'ingredientDetailsReducer';

uncledeni committed:
Complete unit tests for reducers

complete unit tests for reducers (need some check after);
export initial states for 'auth', 'burger-constructor', 'feed-order-details', 'ws-feed-reducer', 'ws-profile-feed-reducer';

uncledeni committed:
Deploy prepare

Use some @ts-ignore to hide some error for normal deploy

uncledeni committed:
Fix Cypress test

mikebars1995 commented on cypress/e2e/spec_profile.cy.ts:
```diff
@@ -0,0 +1,39 @@
+
+  it('should open modal for all ingredients and close it', () => {
+    // переход на страницу "Конструктора"
+    cy.visit('http://localhost:3000/profile');
```
**Нужно исправить**

`http://localhost:3000`

`http://localhost:3000` повторяется несколько раз в коде тестов. Нужно вынести в константу `testUrl`, чтобы можно было одним движением изменить урл для тестов

Базовый урл для тестов прописан в конфиге, нужно использовать его

mikebars1995 commented on cypress.config.ts:
```diff
@@ -0,0 +1,20 @@
+  e2e: {
+    viewportHeight: 1080,
+    viewportWidth: 1920,
+    baseUrl: 'http://localhost:3000/',
```
**Отлично**


* Отлично, что базовый урл вынесен в конфиг

mikebars1995 commented on cypress/e2e/spec.cy.ts:
```diff
@@ -0,0 +1,45 @@
+        
+    // установка моков токенов (из констант, 
+    // падают после просрачивания времени жизни refresh т.к. не происходит обновления) 
+    window.localStorage.setItem("accessToken", aT);
```
**Нужно исправить**

В тестах принято в `beforeEach` устанавливать куки и `localStorage`, а еще их нужно обязательно **удалять** в `afterEach`, чтобы не засорять  куки и `localStorage` тестовыми данными

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная:


* Все юнит-тесты (40 шт) проходят успешно
* Все cypress-тесты проходят успешно
* Код тестов аккуратный и понятный
* Отлично, что базовый урл вынесен в конфиг


но есть некоторые недочеты:


* `http://localhost:3000` повторяется несколько раз в коде тестов. Нужно вынести в константу `testUrl`, чтобы можно было одним движением изменить урл для тестов. Базовый урл для тестов прописан в конфиге, нужно использовать его
* В тестах принято в `beforeEach` устанавливать куки и `localStorage`, а еще их нужно обязательно **удалять** в `afterEach`, чтобы не засорять  куки и `localStorage` тестовыми данными

uncledeni committed:
Tests and deploy fixes

Update 'homepage' in package;
Update cypress tests; use baseUrl instead of string url;
add afterEach block for storage cleaning;
change 'BrowserRouter' to 'HashRouter' in 'index.tsx';
delete all 'opacity: 0' from project style modules;

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
